### PR TITLE
Implement basic authentication functionality

### PR DESCRIPTION
**Pull request by eash130**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      // Prompt user to login to their Google account.
+      String loginLink = userService.createLoginURL("/login");
+      response.getWriter().println("<h3>Welcome!</h3>");
+      response.getWriter().println("<p>Click <a href=\"" + loginLink + "\">here</a> to login.</p>");
+      response.getWriter().println("<p>Click <a href=\"/\">here</a> to return to the home page</p>");
```
Comment (alygerharter) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: perhaps put these blocks into smaller functions for better readability?

```
if (!userService.isUserLoggedIn()) {
   promptLogIn();
} else if (...) {
   etc
```

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    String nickname = request.getParameter(NICKNAME);
+    if (!nickname.isEmpty()) {
+      Entity userEntity = new Entity(USER_INFO);
+      userEntity.setProperty(NICKNAME, nickname);
+      UserService userService = UserServiceFactory.getUserService();
+      userEntity.setProperty(USER_ID, userService.getCurrentUser().getUserId());
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(userEntity);
+    }
+    response.sendRedirect("/login");
```
Comment (alygerharter) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: does this mean it still redirects to login even if the nickname is empty? what does that mean?

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      // Prompt user to login to their Google account.
+      String loginLink = userService.createLoginURL("/login");
+      response.getWriter().println("<h3>Welcome!</h3>");
+      response.getWriter().println("<p>Click <a href=\"" + loginLink + "\">here</a> to login.</p>");
+      response.getWriter().println("<p>Click <a href=\"/\">here</a> to return to the home page</p>");
+    } else if (getUserNickname(userService.getCurrentUser().getUserId()) == "") {
```
Comment (alygerharter) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: use isEmpty() instead of ==""

**Review by alygerharter — CHANGES_REQUESTED** [review]

**Comment by eash130** [comment]
Need to restructure this as it does not work seamlessly with the frontend of the web page. New PR will allow the user to use a login link on the home page rather than navigating to /login

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java (hunk 1 above)
Comment (eash130) [reply portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: Removed in newest commit

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java (hunk 2 above)
Comment (eash130) [reply portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: Removed in newest commit.

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java (hunk 3 above)
Comment (eash130) [reply portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: Removed in newest commit.

**Review by alygerharter — APPROVED** [review]

**Pull request merged**